db/migrate/20200126203714_remove_vim_types_from_ems_events.rb

```diff
@@ -0,0 +1,86 @@
+class RemoveVimTypesFromEmsEvents < ActiveRecord::Migration[5.1]
+  disable_ddl_transaction!
+  include MigrationHelper
+
+  BATCH_SIZE = 5_000
+
+  module VimType
+    def vimType
+      @vimType.nil? ? nil : @vimType.to_s
+    end
+
+    def vimType=(val)
+      @vimType = val.nil? ? nil : val.to_sym
+    end
+
+    def xsiType
+      @xsiType.nil? ? nil : @xsiType.to_s
+    end
+
+    def xsiType=(val)
+      @xsiType = val.nil? ? nil : val.to_sym
+    end
+  end
+
+  class VimHash < Hash; include VimType; end
+  class VimArray < Array; include VimType; end
+  class VimString < String; include VimType; end
+
+  class EventStream < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    vim_hash_backup   = const_replace(:VimHash, VimHash)
+    vim_string_backup = const_replace(:VimString, VimString)
+    vim_array_backup  = const_replace(:VimArray, VimArray)
```

Comment:
Would it be possible to use some yielding method here that `ensure`s that the cleanup occurs?

Picturing this like
```ruby
 def up
   with_replaced_consts(:VimHash => VimHash, :VimString => VimString, :VimArray => VimArray) do
     # stuff
   end
 end

Reply:
:+1: yeah definitely

Reply:
I thikn these backups are now no longer necessary

Reply:
Yeah just forgot to delete them